### PR TITLE
Media upload files ui.

### PR DESCRIPTION
**Pull request by rotemee**
## PR Checklist

- [ ] The commit message follows our guidelines:  https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

*

## Description
An explanation of what is done in this PR

*

## Test instructions
This PR can be tested by following these steps:

*

## Quality assurance

- [ ] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

Fixes #

### CODE REVIEW

File: assets/dev/js/editor/controls/media.js [hunk 1]
```diff
@@ -60,6 +64,7 @@ ControlMediaItemView = ControlMultipleBaseItemView.extend( {
 		this.setValue( {
 			url: '',
 			id: '',
+			fileName: '',
```
Comment (kobizz) [inline assets/dev/js/editor/controls/media.js:67]: We can't insert the filename inside the widget value. You can extract it from the file URL

File: assets/dev/scss/editor/panel/controls/_media.scss [hunk 2]
```diff
@@ -18,12 +18,20 @@
...
+		border-radius: 3px;
+	}
+
 	.eicon-plus-circle {
 		@include absolute-center;
 		color: #fff;
 		font-size: 20px;
 	}
 
+	&-upload-label {
```
Comment (kobizz) [inline assets/dev/scss/editor/panel/controls/_media.scss:31]: remove it

File: assets/dev/scss/editor/panel/controls/_media.scss [hunk 3]
```diff
@@ -138,6 +146,100 @@
...
+
+	// When attachment type is not image and not video
+	&__file {
+		position: relative;
+		display: flex;
+		justify-content: space-between;
+		align-items: center;
+	}
+
+	&__file .elementor-control-upload-content {
```
Comment (kobizz) [inline assets/dev/scss/editor/panel/controls/_media.scss:161]: Nest all nested selectors

File: assets/dev/scss/editor/panel/controls/_media.scss [hunk 4]
```diff
@@ -138,6 +146,100 @@
...
+		padding-left: 5px;
+	}
+
+	&__file &__upload-label {
+		color: #A4AFB7;
```
Comment (kobizz) [inline assets/dev/scss/editor/panel/controls/_media.scss:166]: Use our system colors

File: assets/dev/scss/editor/panel/controls/_media.scss [hunk 5]
```diff
@@ -138,6 +146,100 @@
...
+		font-size: 12px;
+		font-family: Roboto;
```
Comment (kobizz) [inline assets/dev/scss/editor/panel/controls/_media.scss:168]: This value should be inherited from the panel

File: assets/dev/scss/editor/panel/controls/_media.scss [hunk 6]
```diff
@@ -138,6 +146,100 @@
...
+		font-style: italic;
+		font-weight: normal;
```
Comment (kobizz) [inline assets/dev/scss/editor/panel/controls/_media.scss:170]: Same as above

File: assets/dev/scss/editor/panel/controls/_media.scss [hunk 7]
```diff
@@ -138,6 +146,100 @@
...
+	}
+
+	&:not(.elementor-media-empty) &__file &__upload-label {
```
Comment (kobizz) [inline assets/dev/scss/editor/panel/controls/_media.scss:173]: We need to discuss together the rest of this file

**Review by kobizz — CHANGES_REQUESTED** [review]

File: assets/dev/js/editor/controls/media.js [hunk 8]
```diff
@@ -102,7 +106,7 @@ ControlMediaItemView = ControlMultipleBaseItemView.extend( {
 		if ( attachment.url ) {
 			this.setValue( {
 				url: attachment.url,
-				id: attachment.id,
+				id: attachment.id
```
Comment (kobizz) [inline assets/dev/js/editor/controls/media.js:109]: Bring back the comma. We always put a comma even in the last object property

**Review by kobizz — CHANGES_REQUESTED** [review]

File: assets/dev/scss/editor/panel/controls/_media.scss [hunk 9]
```diff
@@ -138,6 +142,100 @@
...
+		display: flex;
+		align-items: center;
+	}
+
+	&.elementor-media-empty &__file-info,
+	&.elementor-media-empty &__remove {
+		display: none;
+	}
+
+	&__file .elementor-control-file-area {
```
Comment (kobizz) [inline assets/dev/scss/editor/panel/controls/_media.scss:183]: Remvoe it

File: assets/dev/scss/editor/panel/controls/_media.scss [hunk 10]
```diff
@@ -138,6 +142,100 @@
...
+		align-items: center;
+		cursor: pointer;
+		display: flex;
+		padding: 15px 10px;
+	}
+
+	&__file .elementor-control-upload-controls {
```
Comment (kobizz) [inline assets/dev/scss/editor/panel/controls/_media.scss:190]: Remove it

File: assets/dev/scss/editor/panel/controls/_media.scss [hunk 11]
```diff
@@ -138,6 +142,100 @@
...
+		top: initial;
+		right: initial;
+		position: relative;
+		border-right: 1px solid $editor-lightest;
+		box-shadow: none;
+	}
+
+	&__file &-upload-button,
+	.elementor-control-media:not(.elementor-media-empty) &__file &-upload-button,
+	&__file &__remove {
```
Comment (kobizz) [inline assets/dev/scss/editor/panel/controls/_media.scss:206]: Check for duplications

File: assets/dev/scss/editor/panel/controls/_media.scss [hunk 12]
```diff
@@ -138,6 +142,100 @@
...
+		display: flex;
+		padding: 15px 10px;
+	}
+
+	&__file .elementor-control-upload-controls {
+		display: flex;
+		border-left: 1px solid $editor-lightest;
+	}
+
+	&__file &__remove {
```
Comment (kobizz) [inline assets/dev/scss/editor/panel/controls/_media.scss:195]: Separate remove buttons style

File: assets/dev/scss/editor/panel/controls/_media.scss [hunk 13]
```diff
@@ -138,6 +142,100 @@
...
+		position: relative;
+		border-right: 1px solid $editor-lightest;
+		box-shadow: none;
+	}
+
+	&__file &-upload-button,
+	.elementor-control-media:not(.elementor-media-empty) &__file &-upload-button,
+	&__file &__remove {
+		width: 27px;
+		height: 27px;
```
Comment (kobizz) [inline assets/dev/scss/editor/panel/controls/_media.scss:208]: Use `$control-unit` variable

File: assets/dev/scss/editor/panel/controls/_media.scss [hunk 14]
```diff
@@ -138,6 +142,100 @@
...
+	}
+
+	&__file &-upload-button:hover {
+		background-color: initial;
+	}
+
+	&__file &__icon,
+	&__file &__file-name {
+		font-family: Roboto;
+		font-style: normal;
```
Comment (kobizz) [inline assets/dev/scss/editor/panel/controls/_media.scss:225]: Should be inherited

File: assets/dev/scss/editor/panel/controls/_media.scss [hunk 15]
```diff
@@ -138,6 +142,100 @@
...
+		position: relative;
+		background-color: initial;
+		display: flex;
+		justify-content: center;
+		align-items: center;
+		color: $editor-dark;
+	}
+
+	&__file &-upload-button:hover {
+		background-color: initial;
```
Comment (kobizz) [inline assets/dev/scss/editor/panel/controls/_media.scss:219]: Apply on media mode only

File: assets/dev/scss/editor/panel/controls/_media.scss [hunk 16]
```diff
@@ -138,6 +142,100 @@
...
+		font-size: 12px;
+		color: $editor-dark;
+	}
+
+	&__file &__icon {
+		margin-right: 5px;
+	}
+
+	&__file &-area {
+		background-image: none;
```
Comment (kobizz) [inline assets/dev/scss/editor/panel/controls/_media.scss:236]: Apply on media mode only

**Review by kobizz — CHANGES_REQUESTED** [review]

File: assets/dev/scss/editor/panel/controls/_media.scss [hunk 17]
```diff
@@ -116,28 +116,113 @@
...
+				font-size: 12px;
+				font-style: italic;
+			}
+
+			&__info {
+				display: flex;
+				align-items: center;
+
+				&__icon,
+				&__name {
```
Comment (kobizz) [inline assets/dev/scss/editor/panel/controls/_media.scss:171]: Instead of that, add these properties to their shared parent

**Review by kobizz — CHANGES_REQUESTED** [review]

**Comment by kobizz** [comment]
Always use direction functions

**Review by kobizz — APPROVED** [review]

**Review by KingYes — APPROVED** [review]

**Pull request merged**